lib/krypto/nacler.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+require "rbnacl"
+require "openssl"
+
+module Krypto
+  class Nacler
+    def initialize(key, counterparty)
+      raise "Missing key" unless key
+      @key = key
+      raise "Missing counter party" unless counterparty
+      @counterparty = counterparty
+    end
+
+    def seal(plaintext)
+      shared = @key.dh_compute_key(@counterparty)
+      shared_key = OpenSSL::Digest::SHA256.digest(shared)
+      box = RbNaCl::SimpleBox.from_secret_key(shared_key)
```

Comment:
If these don't change, they could be in `initialize` and not seal. But at the volume here, it doesn't much matter. (The performance overhead seems fine)

Though it does mean repeating it between seal and open.

Reply:
I'll add that to the next PR!